src/components/ProductPage/ProductPage.jsx

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+import { filterData, useProductContext } from "../../context/productContext/ProductProvider";
+import { productData } from "../../data";
+import ProductCard from "../ProductCard";
+import ProductFilters from "./ProductFilter";
+
+export default function ProductPage() {
+
+    const {state,dispatch} = useProductContext()
+
+
+    return (
+        <>
+            <ProductFilters />
+            <main className="grand-main">
+                <h2 className="font-black">Showing All Products</h2>
+                <div className="products-list">
+                    {filterData(state,productData).map((product) => {
```

Comment:
Suggestion: Do all the filtering in your context only and pass down the filteredData 
instead of getting data and state from context and then passing it to a function.

Reply:
done sudip thanks